src/main/java/org/acme/infinispanclient/service/DataLoader.java

```diff
@@ -0,0 +1,113 @@
+package org.acme.infinispanclient.service;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+
+import org.acme.infinispanclient.model.HPCharacter;
+import org.acme.infinispanclient.model.HPMagic;
+import org.acme.infinispanclient.model.HPSpell;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.quarkus.runtime.StartupEvent;
+
+@ApplicationScoped
+public class DataLoader {
+   public static final String HP_CHARACTERS_NAME = "characters";
+   public static final String HP_SPELLS_NAME = "spells";
+   public static final String HP_MAGIC_NAME = "magic";
+
+   private static final Logger LOGGER = LoggerFactory.getLogger(DataLoader.class.getName());
+
+   @Inject
+   @ConfigProperty(name = "characters.filename")
+   private String charactersFileName;
+
+   @Inject
+   @ConfigProperty(name = "spells.filename")
+   private String spellsFileName;
+
+   @Inject
+   private RemoteCacheManager cacheManager;
+
+   void onStart(@Observes StartupEvent ev) {
```

Comment:
@wburns Did not dig further yet, but I've already experienced the following scenario: when the 3 caches are created for the first time, and then I call "clear", when I try to access to the cache creating the data, caches are not found. It says they don't exist. Even if I have the RemoteCache object. If I retry a second time, caches do exist so the application works properly

Reply:
That is odd, I don't know. It sounds like a server thing though, but as I mentioned I don't know.

Reply:
I read 23:36:06,069 INFO  [org.jboss.as.clustering.infinispan] (async-thread--p3-t1) DGISPN0001: Started characters cache from local container
23:36:06,083 INFO  [org.jboss.as.clustering.infinispan] (async-thread--p3-t2) DGISPN0001: Started spells cache from local container
23:36:06,096 INFO  [org.jboss.as.clustering.infinispan] (async-thread--p3-t3) DGISPN0001: Started magic cache from local container

Async thread, so this is not blocking ... that explains the behaviour

Reply:
ah, and I don't see a way to wait for that.. but I don't have the API in front of me :(

Reply:
So, I'm thinking that before doing @Inject for a RemoteCache in @applicationscoped beans with the @Remote and @Inject, could we imagine something that will allow us to create those caches dynamically before caches get "injected" ?

Reply:
Ok, so I found a way to do it now. I'm waiting 10s before launching the scheduler. Enough time to create caches most of the time. If after that time caches are not there, well, I log messages to ask for a reboot.... Not perfect but enough for now IMO